File: project/hyeok-onboarding/src/test/java/com/INNER_CIRCLE_ICD4/innerCircle/SurveyRepositoryTest.java
package com.INNER_CIRCLE_ICD4.innerCircle;

import com.INNER_CIRCLE_ICD4.innerCircle.repository.*;
import com.INNER_CIRCLE_ICD4.innerCircle.domain.*;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import java.util.UUID;

import static org.assertj.core.api.Assertions.assertThat;

@DataJpaTest
public class SurveyRepositoryTest {

    @Autowired
    private SurveyRepository surveyRepository;

    @Autowired
    private QuestionRepository questionRepository;

    @Autowired
    private ChoiceRepository choiceRepository;

    @Test
    void surveyRepositoryTestFunc() {
        // 1. 설문 생성
        Survey survey = new Survey("테스트 설문", "설문 설명입니다.");
        surveyRepository.save(survey);

        // 2. 질문 생성
        Question question = new Question("질문 제목", "질문 설명", QuestionType.SINGLE_CHOICE, true);
        question.setSurvey(survey);
        questionRepository.save(question);

        // 3. 선택지 생성 및 추가
        Choice choice1 = new Choice("선택지1", 0);
        choice1.setQuestion(question);
        Choice choice2 = new Choice("선택지2", 1);
        choice2.setQuestion(question);

        choiceRepository.save(choice1);
        choiceRepository.save(choice2);

        // 4. 저장 후 조회
        Survey foundSurvey = surveyRepository.findById(survey.getId()).orElseThrow();
        assertThat(foundSurvey.getTitle()).isEqualTo("테스트 설문");
    }
}

File: project/jihochoi-onboarding/QueryBox/src/test/kotlin/com/wlghsp/querybox/repository/SurveyRepositoryTest.kt
package com.wlghsp.querybox.repository

import com.wlghsp.querybox.domain.survey.Option
import com.wlghsp.querybox.domain.survey.Options
import com.wlghsp.querybox.domain.survey.Question
import com.wlghsp.querybox.domain.survey.QuestionType
import com.wlghsp.querybox.domain.survey.Questions
import com.wlghsp.querybox.domain.survey.Survey
import jakarta.persistence.EntityManager
import org.assertj.core.api.AssertionsForInterfaceTypes.assertThat
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest

@DataJpaTest
class SurveyRepositoryTest @Autowired constructor(
    val surveyRepository: SurveyRepository,
    val entityManager: EntityManager,
) {

    @DisplayName("설문과 항목들을 함께 조회한다.")
    @Test
    fun findSurveyWithQuestionsById_loadsQuestions() {
        val question = Question(
            name = "좋아하는 언어는?",
            description = "언어를 선택하세요",
            type = QuestionType.SINGLE_CHOICE,
            required = true,
            options = Options.of(listOf(Option("Kotlin"), Option("Java")))
        )

        val survey = Survey.of(
            title = "개발자 설문",
            description = "개발자에게 묻습니다.",
            questions = Questions.of(listOf(question))
        ).apply {
            addQuestion(question)
        }

        surveyRepository.save(survey)
        entityManager.flush()
        entityManager.clear()

        val foundSurvey = surveyRepository.findSurveyWithQuestionsById(survey.id)

        assertThat(foundSurvey).isNotNull
        assertThat(foundSurvey!!.questions.values()).hasSize(1)
        assertThat(foundSurvey.questions.values()[0].name).isEqualTo("좋아하는 언어는?")
    }

}